include/nbl/builtin/hlsl/device_capabilities_traits.hlsl

```diff
@@ -7,15 +7,1323 @@
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[1], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxWorkgroupSize[2], uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(subPixelPrecisionBits, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(subTexelPrecisionBits, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(mipmapPrecisionBits, uint8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxDrawIndirectCount, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerLodBias, float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(maxSamplerAnisotropyLog2, uint8_t, 0.0f);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxViewports, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewportDims[0], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxViewportDims[1], uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(viewportBoundsRange[0], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(viewportBoundsRange[1], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(viewportSubPixelBits, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(bufferViewAlignment, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(minUBOAlignment, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(minSSBOAlignment, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(minTexelOffset, int8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTexelOffset, int8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(minTexelGatherOffset, int8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxTexelGatherOffset, int8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(minInterpolationOffset, float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(maxInterpolationOffset, float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(subPixelInterpolationOffsetBits, uint8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferWidth, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferHeight, uint32_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxFramebufferLayers, uint32_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxColorAttachments, uint8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxSampleMaskWords, uint8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(timestampPeriodInNanoSeconds, float, 0.0f);
+
+NBL_GENERATE_GET_OR_DEFAULT(maxClipDistances, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxCullDistances, uint8_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(maxCombinedClipAndCullDistances, uint8_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(pointSizeRange[0], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(pointSizeRange[1], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(lineWidthRange[0], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(lineWidthRange[1], float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(pointSizeGranularity, float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(lineWidthGranularity, float, 0.0f);
+NBL_GENERATE_GET_OR_DEFAULT(strictLines, bool, false);
+
+NBL_GENERATE_GET_OR_DEFAULT(optimalBufferCopyOffsetAlignment, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(optimalBufferCopyRowPitchAlignment, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(nonCoherentAtomSize, uint16_t, 0);
+
+NBL_GENERATE_GET_OR_DEFAULT(subgroupSize, uint16_t, 0);
+NBL_GENERATE_GET_OR_DEFAULT(subgroupOpsShaderStages, core::bitflag<asset::IShader::E_SHADER_STAGE, 0);
```

Comment:
obviously there's no bitflag on hlsl, so use a plain  `uint8_t`

Reply:
Done

Reply:
Also I lied to you, there's no `uint8_t` in HLSL, only `uint16_t`

But there are more than 8 shader stages anyway....